### ISSUE TITLE
Multiple users

### ISSUE THREAD

**Issue by khanmo97** [opensourcepos/opensourcepos#2976]
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [x] Reporting an issue of an unmodified OSPOS installation
- [x] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [x] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [x] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [x] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [x] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [x] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Installation information

- OSPOS version is: 
- OSPOS git commit hash is: 
- PHP version is: 
- MySQL or MariaDB version is: 
- OS and version is: 
- WebServer is: 
- Selected language is: 
- (If applicable) Docker installation: 
- (If applicable) Installation package for the LAMP/LEMP stack is: 

### Issue / Bug  / Question / New Feature

Do you guys think we could introduce a new feature? This feature would hide sensitive information, in that way Business owners do not have to worry about employees snooping around. If you guys allow it, could I work on a PR for this? Probably will take some good time though.

**Comment by daN4cat** [comment]
New features welcomed and go for a PR.

I guess what you are referring is to expand the privilege granting we have at the moment but maybe by adding roles. Use a role based access control.

**Comment by khanmo97** [comment]
Yah exactly. That would be a good feature to add. A user role based account. Thanks man!

**Comment by jekkos** [comment]
Ok so with 'role' you mean a predefined set of 'grants' in this case that you assign to a user?

**Comment by jekkos** [comment]
Like in this case as an example: role cashier, role inventory manager, role admin?
An easy way to implement this would be by defining what permissions a role consists of in the code and then you don't need to change anything in the datamodel. If a user has that role, you just give it all the grants to the permissions that this role has assigned.

**Comment by jekkos** [comment]
In the UI you can then 

* add a dropdown (only visible to admin) in the employee edit form
* fill it the dropdown with the roles you have defined in an array
* preselect the permissions inside that role once the admin selects a role from this dropdown

**Comment by daN4cat** [comment]
Luckily we never went for Access Control Lists, the old group concept coming with AD.
We are quite granular and assign per user the singular access.
With roles we define the set of permissions and allocate roles to users.

https://en.m.wikipedia.org/wiki/Role-based_access_control

**Comment by daN4cat** [comment]
Yes, this will be definitely a great addition for the Authorization model.

When I find the time I want to separate the Authentication and support OpenID connect with external services like Google.
But we need a means to control that.

**Comment by khanmo97** [comment]
> Yes, this will be definitely a great addition for the Authorization model.
> 
> When I find the time I want to separate the Authentication and support OpenID connect with external services like Google.
> But we need a means to control

Are you saying connect/login via google? That'll be tough to do, especially to keep it free. Can you guys recommend a good debugger for php?

**Comment by jekkos** [comment]
I'd go for xdebug. it works pretty well with phpstorm. If you want an easy setup you can just use the dev docker-compose in this proejct (it has xdebug preinstalled). I mostly edit my code and then spin up the container that then reflects the changes immediately.

**Comment by midsoft2010** [comment]
Like i have a problem with employees seeing my profits and at the same time i want them to know the total sales of the day and items sold in a day.. i don't like the idea they are able to filter by the date and view my profits of the month or of the year
